src/extensions/old/averagewaveformsold/AverageWaveformsNew.tsx

```diff
@@ -22,11 +22,19 @@ const AverageWaveformsNew: React.FunctionComponent<SortingViewProps> = (props) =
     // }, [hither, setResult, calcMode, setCalcMode])
     // return <div>AAAA: {calcMode} {result + ''}</div>
     const selectedUnitIdsLookup: {[key: string]: boolean} = (selection.selectedUnitIds || []).reduce((m, uid) => {m[uid + ''] = true; return m}, {} as {[key: string]: boolean})
+    const handleUnitClicked = useCallback((unitId: number, event: {ctrlKey?: boolean, shiftKey?: boolean}) => {
+        selectionDispatch({
+            type: 'UnitClicked',
+            unitId,
+            ctrlKey: event.ctrlKey,
+            shiftKey: event.shiftKey
+        })
+    }, [selectionDispatch])
```

Comment:
I'd be nice to centralize this a bit more if it's possible to do so, I see it's repeated over a couple different files.

Actually I'm not too clear on the distinction between `src/extensions/old` and `src/extensions/` and what is & isn't current. Some of this may have been superseded by later PRs. Are we due for a little New Year cleaning in terms of unused files & commented code?

Reply:
Yeah, maybe a `useHandleUnitClicked(selectionDispatch)` custom hook

The plan is for `old/` stuff to go away at some point, but keeping it in for now to avoid breaking and for reference (e.g., bokeh stuff).